compiler/luci-interpreter/src/kernels/InstanceNorm.cpp

```diff
@@ -63,58 +82,112 @@ void InstanceNorm::evalFloat() const
 {
   float activation_min, activation_max;
   calculateActivationRange(params().activation, &activation_min, &activation_max);
-  auto input_shape = getTensorShape(input());
+  tflite::RuntimeShape input_shape = getTensorShape(input());
   auto output_shape = getTensorShape(output());
-  const int32_t batches = tflite::MatchingDim(input_shape, 0, output_shape, 0);
-  const int32_t heights = tflite::MatchingDim(input_shape, 1, output_shape, 1);
-  const int32_t widths = tflite::MatchingDim(input_shape, 2, output_shape, 2);
-  const int32_t channels = tflite::MatchingDim(input_shape, 3, output_shape, 3);
-  const float *input_data = getTensorData<float>(input());
-  const float *gamma_data = getTensorData<float>(gamma());
-  auto gamma_shape = getTensorShape(gamma());
-  bool single_gamma = gamma_shape.DimensionsCount() == 1 && gamma_shape.Dims(0) == 1;
-  const float *beta_data = getTensorData<float>(beta());
-  auto beta_shape = getTensorShape(beta());
-  bool single_beta = beta_shape.DimensionsCount() == 1 && beta_shape.Dims(0) == 1;
-  float *output_data = getTensorData<float>(output());
-  for (int32_t batch = 0; batch < batches; batch++)
+
```

Comment:
i think the difference between 4D and 3D is the content of dimension, 
so It would be better to initialize only the related contents inside the conditional code (if(4), if(3))
and the rest of initialization code can be moved here.
It seems to remove duplicate codes in the initialization code.
```suggestion
  float activation_min, activation_max;
  calculateActivationRange(params().activation, &activation_min, &activation_max);
  tflite::RuntimeShape input_shape = getTensorShape(input());
  auto output_shape = getTensorShape(output());
  const float *input_data = getTensorData<float>(input());
  const float *gamma_data = getTensorData<float>(gamma());
  auto gamma_shape = getTensorShape(gamma());
  bool single_gamma = gamma_shape.DimensionsCount() == 1 && gamma_shape.Dims(0) == 1;
  const float *beta_data = getTensorData<float>(beta());
  auto beta_shape = getTensorShape(beta());
  bool single_beta = beta_shape.DimensionsCount() == 1 && beta_shape.Dims(0) == 1;
  float *output_data = getTensorData<float>(output());

  if (input_shape.DimensionsCount() == 4)
  {
    // Dimensions for image case are (N x H x W x C)
    const int32_t batches = tflite::MatchingDim(input_shape, 0, output_shape, 0);
    const int32_t heights = tflite::MatchingDim(input_shape, 1, output_shape, 1);
    const int32_t widths = tflite::MatchingDim(input_shape, 2, output_shape, 2);
    const int32_t channels = tflite::MatchingDim(input_shape, 3, output_shape, 3);
    for (int32_t batch = 0; batch < batches; batch++)
    {      
```

Reply:
Thanks, I modified the code as you suggested.